tests/integration.rs

```diff
@@ -275,12 +278,57 @@ fn non_existent(#[case] harness: Harness) {
 	);
 }
 
-//#[apply(#[case] all_images)]
-//fn readlink_long(#[case] harness: Harness) {
-//	let d = &harness.d;
-//
-//	let link = std::fs::read_link(d.path().join("long-link")).unwrap();
-//	let expected = (0..200).map(|_| "./").fold(String::new(), |a, x| a + x) + "/file1";
-//
-//	assert_eq!(link, Path::new(&expected));
-//}
+// This tests both sparse files and 2nd level indirect block addressing
+#[apply(all_images)]
+fn sparse(#[case] harness: Harness) {
+	let d = &harness.d;
+
+	let mut file = File::open(d.path().join("sparse")).unwrap();
+	let st = file.metadata().unwrap();
+
+	assert_eq!(st.blocks(), 320);
+	assert_eq!(st.size(), 134643712);
+
+	file.seek(SeekFrom::Start((12 + 4096) * 32768)).unwrap();
+	let mut buf = [0u8; 32768];
+	file.read_exact(&mut buf).unwrap();
+	let expected = [b'x'; 32768];
+	assert_eq!(buf[0], expected[0]);
```

Comment:
Do this.  Otherwise, you're only verifying one byte.
```suggestion
	assert_eq!(buf, expected);
```

Reply:
Thanks, I forgot to change it back. I used `[0]` because I didn't want to hammer my entire terminal when I ran `cargo test`